RFCs/0025-graph-data-model.md

```diff
@@ -54,51 +63,99 @@ Which is similarly extracted as variables:
-The following diagram illustrates the model:
+For the graph data type, we model something very similar. A graph is a collection of *vertices* and *edges*, 
+where edges connect pairs of vertices and can be directed or undirected.
+Vertices and edges have *labels* (similar to the attribute names in struct). 
+There is also a **value at** each vertex and edge, which can be any PartiQL value.
+
+The following example illustrates the model:
 
 ![Example Graph](./0025-graph-data-model/graph.png)
 
 In the above, we have a graph with three vertices, two labeled **`a`** and one labeled **`b`**. We have three edges, one
 labeled **`x`** and two labeled **`y`**. The relationships of the edges to their respective vertices are fully contained
-within the graph. All values within the graph, either an edge or vertex, can be *any* PartiQL value. This generalization
+within the graph. All values within the graph, either at an edge or at a vertex, can be *any* PartiQL value. This generalization
 is consistent with the other container types and fits nicely in PartiQL’s data model. This also means that PartiQL
 graphs *could* have vertices or edges that themselves be graphs, and likewise values can be as simple as scalar values.
 
-As a common example, let’s consider how we could model a *labeled property graph (LPG)* to PartiQL. For LPG, each vertex
-and edge are a property bag which is a struct in PartiQL.
+As an important example, *labeled property graphs (LPG)* of the GPML paper[^2] are modeled 
+straightforwardly in PartiQL: a value at each vertex and edge is restricted to be a PartiQL struct.
 
+<!-- TODO: The following RDF example may need adjustments to fit with the finalized data model.
+     E.g., NULL labels may have to become empty label sets.  -->
 Similarly, Resource Description Framework (RDF)[^1] graphs could be modeled in PartiQL by having non-literal, non-blank
 vertices and edges labeled by URI strings with their values being NULL. RDF literals could be a NULL labeled node with
 their value being any corresponding PartiQL value (this is a generalization of RDF as literals are only strings in RDF’s
 model). RDF blank nodes can be denoted with a label that never conflicts with URI such as **`_:my_blank`** (**`_`** is
 never a valid scheme for a URI).
 
-Even though PartiQL generalizes the graph data model, it is *not required* that a database actually exposes arbitrary
-values at vertex or edge values, this is similar to PartiQL over a relational table, where attributes of a row are
+Even though PartiQL defines a very general graph data model, it is *not required* that a host database actually supports arbitrary
+values at vertices or edges. This is similar to PartiQL over a relational database, where attributes of a row are
 restricted to scalars.
 
-## Data Model Integration with PartiQL Query
+## Addition to the PartiQL data model
```

Comment:
*Minor*: Should this and other section headings be in [title case][1]?

[1]: https://en.wikipedia.org/wiki/Title_case